src/lib/legacy/Zikula/Doctrine2/Entity/Category.php

```diff
@@ -150,7 +150,7 @@ public function getParent()
         return $this->parent;
     }
 
-    public function setParent(Zikula_Doctrine2_Entity_Category $parent)
+    public function setParent(self $parent)
     {
```

Comment:
is this a php > 5.5 thing? I'm not sure we should be supporting this?

Reply:
I do not know, too.

Reply:
Seems like it is possible since PHP 5: https://blog.madewithlove.be/post/self-and-parent-type-hints/